From c91948dedaaecdde827a1898409210d2d6a96e76 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:05:04 -0400
Subject: [fs] btrfs: use E2BIG instead of EIO if compression does not help

Message-id: <1407179109-13714-418-git-send-email-sandeen@redhat.com>
Patchwork-id: 87289
O-Subject: [RHEL7.1 PATCH 417/422] btrfs: use E2BIG instead of EIO if compression does not help
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: David Sterba <dsterba@suse.cz>

commit 130d5b415a091e493ac1508b9d27bbb85ba7b8c0
Author: David Sterba <dsterba@suse.cz>
Date:   Fri Jun 20 11:43:20 2014 +0200

    btrfs: use E2BIG instead of EIO if compression does not help

    Return codes got updated in 60e1975acb48fc3d74a3422b21dde74c977ac3d5
    (btrfs: return errno instead of -1 from compression)
    lzo wrapper returns E2BIG in this case, do the same for zlib.

    Signed-off-by: David Sterba <dsterba@suse.cz>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/zlib.c b/fs/btrfs/zlib.c
index 4f19631..b67d8fc 100644
--- a/fs/btrfs/zlib.c
+++ b/fs/btrfs/zlib.c
@@ -136,7 +136,7 @@ static int zlib_compress_pages(struct list_head *ws,
   if (workspace->def_strm.total_in > 8192 &&
       workspace->def_strm.total_in <
       workspace->def_strm.total_out) {
-   ret = -EIO;
+   ret = -E2BIG;
    goto out;
   }
   /* we need another page for writing out.  Test this
-- 
1.7.1